service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -541,6 +532,22 @@ private ApiCloneControlledGcpBigQueryDatasetResult fetchCloneBigQueryDatasetResu
         result, ControllerUtils.getAsyncResponseCode(result.getJobReport()));
   }
 
+  /**
+   * Extract the assigned user from a request to create a controlled resource. This field is only
+   * populated for private resources, but if a resource is private then a "null" value means "assign
+   * this resource to the requesting user" and should be populated here.
+   */
+  private String assignedUserFromBodyOrToken(
+      ApiControlledResourceCommonFields commonFields, AuthenticatedUserRequest userRequest) {
+    if (commonFields.getAccessScope() != ApiAccessScope.PRIVATE_ACCESS) {
+      return null;
+    }
+    return Optional.ofNullable(commonFields.getPrivateResourceUser().getUserName())
+        .orElse(
```

Comment:
use `orElseGet` to avoid the Sam call in the case where it's not necessary

Reply:
huh, I had always assumed the argument to `orElse` was only evaluated if the optional was null. TIL, thanks!